bundles/org.openhab.binding.myuplink/src/main/java/org/openhab/binding/myuplink/internal/command/AbstractWriteCommand.java

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.myuplink.internal.command;
+
+import static org.openhab.binding.myuplink.internal.MyUplinkBindingConstants.EMPTY;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jetty.client.api.Request;
+import org.openhab.binding.myuplink.internal.Utils;
+import org.openhab.binding.myuplink.internal.handler.MyUplinkThingHandler;
+import org.openhab.binding.myuplink.internal.model.ValidationException;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * base class for all write commands. common logic should be implemented here
+ *
+ * @author Alexander Friese - initial contribution
+ */
+@NonNullByDefault
+public abstract class AbstractWriteCommand extends AbstractCommand {
+    private final Logger logger = LoggerFactory.getLogger(AbstractWriteCommand.class);
+
+    protected final Channel channel;
+    protected final Command command;
+
+    /**
+     * the constructor
+     */
+    public AbstractWriteCommand(MyUplinkThingHandler handler, Channel channel, Command command,
+            RetryOnFailure retryOnFailure, ProcessFailureResponse processFailureResponse,
+            JsonResultProcessor resultProcessor) {
+        super(handler, retryOnFailure, processFailureResponse, resultProcessor);
+        this.channel = channel;
+        this.command = command;
+    }
+
+    /**
+     * helper method for write commands that extracts value from command.
+     *
+     * @return value as String without unit.
+     */
+    protected String getCommandValue() {
+        if (command instanceof QuantityType<?> quantityCommand) {
+            // this is necessary because we must not send the unit to the backend
+            return String.valueOf(quantityCommand.doubleValue());
```

Comment:
good point. Do you have any example how to achieve this conversion? I think this will be temperatures in most cases.

Reply:
quantityCommand.toUnit(Units.Celcius)

Reply:
I changed the code. Since this is a generic binding it is hard to cover all possible cases.
I am unsure if it is better to throw an exception if we have a type that is not explicitely covered ot log it and assume no type conversion is needed. I implemented the second variant since this will at least work in a basic way if the user did not apply any type conversions.

Reply:
Could you point me to an example that i hard to cover? I would assume the remote service has the expected unit documented for all api calls?!

Reply:
Well the API itself is generic. Depending on the actual hardware it returns differnt datapoints with different units. MAybe it is possible to derive an expected unit from the actual unit. I just need to find a way to store it somewhere.